src/transactionIntentsFactories/smartContractTransactionIntentsFactory.spec.ts

```diff
@@ -0,0 +1,108 @@
+        let abiJson = await promises.readFile("src/testdata/adder.abi.json", { encoding: "utf8" });
+        let abiObj = JSON.parse(abiJson);
+        abiRegistry = AbiRegistry.create(abiObj);
+
+        abiSmartContractIntentsFactory = new SmartContractTransactionIntentsFactory({ chainID: "D", minGasLimit: 50000, gasLimitPerByte: 1500 }, abiRegistry);
+    });
+
+    it("should throw error", async function () {
+        const sender = Address.fromBech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th");
+        const gasLimit = 6000000;
+        const args = [0];
+        try {
+            smartContractIntentsFactory.createTransactionIntentForDeploy(sender, adderByteCode.valueOf(), gasLimit, args);
+        }
+        catch (err) {
+            expect(err.message).to.equal("Can't convert args to TypedValues");
+        }
+    });
+
+    it("should build deploy intent", async function () {
+        const sender = Address.fromBech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th");
+        const gasLimit = 6000000;
+        const args = [new U32Value(0)];
+
+        const deployIntent = smartContractIntentsFactory.createTransactionIntentForDeploy(sender, adderByteCode.valueOf(), gasLimit, args);
+        const abiDeployIntent = abiSmartContractIntentsFactory.createTransactionIntentForDeploy(sender, adderByteCode.valueOf(), gasLimit, args);
+
+        assert.equal(deployIntent.sender, "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th");
+        assert.equal(deployIntent.receiver, CONTRACT_DEPLOY_ADDRESS);
+        assert.isDefined(deployIntent.data);
+
+        if (deployIntent.data) {
+            const expectedGasLimit = 6000000 + 50000 + 1500 * deployIntent.data.length;
+            assert.equal(deployIntent.gasLimit.valueOf(), expectedGasLimit);
+        }
+        assert.equal(deployIntent.value, 0);
+
+        assert.deepEqual(deployIntent, abiDeployIntent);
+    });
+
+    it("should build execute intent", async function () {
+        const sender = Address.fromBech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th");
+        const contract = Address.fromBech32("erd1qqqqqqqqqqqqqpgqhy6nl6zq07rnzry8uyh6rtyq0uzgtk3e69fqgtz9l4");
+        const func = "add";
+        const gasLimit = 6000000;
+        const args = [new U32Value(7)];
+
+        const deployIntent = smartContractIntentsFactory.createTransactionIntentForExecute(sender, contract, func, gasLimit, args);
+        const abiDeployIntent = abiSmartContractIntentsFactory.createTransactionIntentForExecute(sender, contract, func, gasLimit, args);
+
+        assert.equal(deployIntent.sender, "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th");
+        assert.equal(deployIntent.receiver, "erd1qqqqqqqqqqqqqpgqhy6nl6zq07rnzry8uyh6rtyq0uzgtk3e69fqgtz9l4");
+
+        assert.isDefined(deployIntent.data);
+        let decoder = new TextDecoder();
+        assert.equal(decoder.decode(deployIntent.data), "add@07");
```

Comment:
`assert.equal(deployIntent.data.toString(), "add@07")` or `assert.equal(deployIntent.data, Buffer.from("add@07"))` should work, without using a `TextDecoder`.

Reply:
Indeed. Works when using `assert.deepEqual()`